repondeur/zam_repondeur/models/amendement.py

```diff
@@ -29,109 +19,206 @@
-    Column("subdiv_jaune", Text, nullable=True),  # éléments de langage
-    #
-    # Numéro de l'amendement
-    #
-    Column("num", Integer, primary_key=True),
-    #
-    # Numéro de révision de l'amendement
-    #
-    Column("rectif", Integer, nullable=False, default=0),
-    #
-    # Auteur de l'amendement
-    #
-    Column("auteur", Text, nullable=True),
-    Column("matricule", Text, nullable=True),
-    Column("groupe", Text, nullable=True),  # groupe parlementaire
-    #
-    # Date de dépôt de l'amendement (est-ce la date initiale,
-    # ou bien est-ce mis à jour si l'amendement est rectifié ?)
-    #
-    Column("date_depot", Date, nullable=True),
-    #
-    Column("sort", Text, nullable=True),  # retiré, adopté, etc.
-    #
-    # Ordre et regroupement lors de la discussion
-    #
-    Column("position", Integer, nullable=True),
-    Column("discussion_commune", Integer, nullable=True),
-    Column("identique", Boolean, nullable=True),
-    Column("parent_num", Integer, ForeignKey("amendements.num"), nullable=True),
-    Column("parent_rectif", Integer, nullable=True),
-    #
-    # Contenu de l'amendement
-    #
-    Column("dispositif", Text, nullable=True),  # texte de l'amendement
-    Column("objet", Text, nullable=True),  # motivation
-    Column("resume", Text, nullable=True),  # résumé de l'objet
-    #
-    # Avis et réponse
-    #
-    Column("avis", Text, nullable=True),  # position du gouvernemnt
-    Column("observations", Text, nullable=True),
-    Column("reponse", Text, nullable=True),
-    Column("comments", Text, nullable=True),
-    #
-    # Informations additionnelles
-    #
-    Column("bookmarked_at", DateTime, nullable=True),
-)
-
-
-mapper(
-    Amendement,
-    amendements_table,
-    properties={
-        # for some reason, those fields do not get mapped automatically,
-        # so we map them explicitly here
-        "subdiv_mult": amendements_table.c.subdiv_mult,
-        "subdiv_pos": amendements_table.c.subdiv_pos,
-        "subdiv_titre": amendements_table.c.subdiv_titre,
-        "subdiv_contenu": amendements_table.c.subdiv_contenu,
-        "subdiv_jaune": amendements_table.c.subdiv_jaune,
-        "alinea": amendements_table.c.alinea,
-        "num": amendements_table.c.num,
-        "rectif": amendements_table.c.rectif,
-        "parent_num": amendements_table.c.parent_num,
-        "parent_rectif": amendements_table.c.parent_rectif,
-        "parent": relationship(
-            Amendement,
-            primaryjoin=(
-                amendements_table.c.parent_num == foreign(amendements_table.c.num)
-            ),
-            uselist=False,  # many-to-one relationship
-        ),
-        "auteur": amendements_table.c.auteur,
-        "groupe": amendements_table.c.groupe,
-    },
-)
+class Amendement(Base):  # type: ignore
+    __tablename__ = "amendements"
+
+    pk = Column(Integer, primary_key=True)
+
+    # Meta informations.
+    num = Column(Integer)
```

Comment:
I did it at the alembic/migrations level, not sure how to do it on a class-based model.

Reply:
It think the way to do this would be to add some *table args* to the model class:
```python
class Amendement(Base):
    __tablename__ = "amendements"
    __table_args__ = (
        UniqueConstraint("num", "lecture_pk"),
    )
```
See: http://docs.sqlalchemy.org/en/latest/orm/extensions/declarative/table_config.html#table-configuration

Reply:
I added the "rectif" within the constraint too.